### PR TITLE
Add basic test setup

### PR DESCRIPTION
**Pull request by johannespfeiffer**
Basic setup to run tests and one simple testcase.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,18 +2,17 @@
...
-    "start": "node dist/server.js",
-    "debug": "node --nolazy --inspect=0.0.0.0:9559 dist/server.js",
-    "test": "npm run lint && echo \"Error: no test specified\" && exit 1",
-    "transpile": "tsc && echo \"--ALL FILES TRANSPILED!\"",
-    "transpile-watch": "tsc --watch",
+    "build": "tsc && cp -r src/templates dist/src",
+    "start": "node dist/src/server.js",
+    "debug": "node --nolazy --inspect=0.0.0.0:9559 dist/src/server.js",
+    "test": "npm run lint && npm run build && mocha --recursive 'dist/tests/**/*.spec.js'",
+    "cover": "nyc npm run test",
```
Comment (teodorpatras) [inline package.json]: what do you think of changing it to:

`"coverage": "nyc --check-coverage=false npm run test"`?

This way we can always run `coverage` (including the CI) to get the latest report, plus we won't have to deal with the non-zero status code that `nyc` returns when coverage is below a certain threshold.

Comment (johannespfeiffer) [reply package.json]: "cover" -> "coverage" 👍
But I actually like the non-zero status code, that way new code should not lower the coverage. Probably it only makes sense if the coverage will be a bit higher. Disabled it for now, as you suggested, see https://github.com/kontist/mock-solaris/pull/5/commits/99d694edd9b15ea3aba7bf989e8eb4eb0e4ef602

Comment (teodorpatras) [reply package.json]: Thanks! I totally agree with coverage breaking a certain build pipeline, but I think that should be enabled once we reach a certain coverage percentage and want to maintain it. For now, at 6.x%, it doesn't really make much sense.

**Review by teodorpatras — APPROVED** [review]
👍

**Pull request merged**